blockchain/network.py

```diff
@@ -670,6 +688,41 @@ def create_node_from_peer(peer):
 
         return map(create_node_from_peer, self.connection_manager.connections)
 
+    def get_unsent_transfer_ids(self, transfer_to=None, verification_id=None):
+        """ retrieve unsent transfer record info (data used for querying block_verification database) """
+        unsent_transfer_ids = []
+        try:
+            logger().info("Retrieving unsent transfer ids...")
+            for transfer_record in vr_transfers_db.get_unsent_verification_records(transfer_to, verification_id):
+                unsent_transfer_ids.append(transfer_record['verification_id'])
+        except:
+            logger().warning("An SQL error has occurred.")
+
+        return unsent_transfer_ids
+
+    def transfer_data(self, received, unreceived):
+        """ mark verifications as sent and return unsent verifications """
+        verifications = []
+        node_unreceived = []
+        all_guids = received + unreceived
+
+        # retrieve ids that calling node is claiming as unreceived
+        for guid in unreceived:
```

Comment:
since the guids are already the primary keys, there's no need to fetch them from the transfer table first just to return the same guid back. You can take this call out.